amethyst_renderer/src/pass/util.rs

```diff
@@ -300,6 +325,80 @@ pub(crate) fn draw_mesh(
     effect.clear();
 }
 
+pub(crate) fn draw_sprite(
+    encoder: &mut Encoder,
+    effect: &mut Effect,
+    sprite_render: &SpriteRender,
+    sprite_sheet_storage: &AssetStorage<SpriteSheet>,
+    tex_storage: &AssetStorage<Texture>,
+    material_texture_set: &MaterialTextureSet,
+    camera: Option<(&Camera, &GlobalTransform)>,
+    global: Option<&GlobalTransform>,
+) {
+    if global.is_none() {
+        return;
+    }
+
+    let sprite_sheet = sprite_sheet_storage.get(&sprite_render.sprite_sheet);
+    if sprite_sheet.is_none() {
+        warn!(
+            "Sprite sheet not loaded for sprite_render: `{:?}`.",
+            sprite_render
+        );
+        return;
+    }
+    let sprite_sheet = sprite_sheet.unwrap();
+
+    let texture_handle = material_texture_set.handle(sprite_sheet.texture_id);
+    if texture_handle.is_none() {
+        warn!(
+            "Texture handle not found for texture id: `{}`.",
+            sprite_sheet.texture_id
+        );
+        return;
+    }
+
+    let texture = tex_storage.get(&texture_handle.unwrap());
+    if texture.is_none() {
+        warn!(
+            "Texture not loaded for texture id: `{}`.",
+            sprite_sheet.texture_id
+        );
```

Comment:
Why this warning?

Reply:
At the time it was, "nothing's rendering, let me add diagnostics to everything that could be missing", and texture followed the pattern of "this is an `Option<..>`, let's log it in case it's `None`". Admittedly if `loader.load(path, PngFormat, ..)` is used (which I suspect it will be most of the time), it wouldn't be `None`.

That said, I'm usually happier to have something that tells me something's missing, even if it's super unlikely.

Reply:
Actually, loader.load will return immediately, even when the texture is not loaded yet.
That means you can have a couple frames without the loaded texture.

Reply:
You should be using `Progress` to track asset loads before using them on entities for this reason.